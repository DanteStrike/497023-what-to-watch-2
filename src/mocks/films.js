export const films = [
  {
    id: 1,
    name: `filmOne`,
    posterImage: `img/filmOne-poster.jpg`,
    preview: {
      image: `img/filmOne.jpg`,
      videoSrc: `https://some-linkOne`
    },
    background: {
      image: `img/filmOne.jpg`,
      color: `#ffffff`
    },
    videoSrc: `https://some-linkOne`,
    description: `Some text filmOne`,
    rating: 1.0,
    scoresCount: 1,
    director: `directorOne`,
    starring: [`actorOne`, `actorTwo`],
    runTime: 1,
    genre: `genreOne`,
    released: 2019,
    isFavorite: false,
  }, {
    id: 3,
    name: `filmTwo`,
    posterImage: `img/filmTwo-poster.jpg`,
    preview: {
      image: `img/filmTwo.jpg`,
      videoSrc: `https://some-linkTwo`
    },
    background: {
      image: `img/filmTwo.jpg`,
      color: `#000000`
    },
    videoSrc: `https://some-linkTwo`,
    description: `Some text filmTwo`,
    rating: 2.0,
    scoresCount: 2,
    director: `directorTwo`,
    starring: [`actorThree`, `actorFour`],
    runTime: 2,
    genre: `genreTwo`,
    released: 2020,
    isFavorite: true,
  }
];
