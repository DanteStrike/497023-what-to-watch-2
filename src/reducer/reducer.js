
const filterFilmsByGenre = (films, genre) => {
  return films.filter((film) => film.genre === genre);
};

const ActionCreator = {
  setGenreFilter: (genre) => ({
    type: `SET_GENRE_FILTER`,
    payload: genre
  }),

  filterFilms: (films, genre) => {
    return {
      type: `SET_FILTERED_FILMS`,
      payload: genre !== `All genre` ? filterFilmsByGenre(films, genre) : films.slice()
    };
  }
};

const initialState = {
  filterGenre: ``,
  filteredFilms: []
};

const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case `SET_GENRE_FILTER`:
      return Object.assign({}, state, {
        filterGenre: action.payload
      });
    case `SET_FILTERED_FILMS`:
      return Object.assign({}, state, {
        filteredFilms: action.payload
      });
  }

  return state;
};

export {reducer, filterFilmsByGenre, ActionCreator};

