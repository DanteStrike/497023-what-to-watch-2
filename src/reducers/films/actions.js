import types from "./types.js";
import utils from "./utils.js";


const loadFilms = (filmsRAW) => {
  return {
    type: types.LOAD_FILMS,
    payload: utils.transformFilmsRAW(filmsRAW)
  };
};

const loadPromo = (PromoRAW) => {
  return {
    type: types.LOAD_PROMO,
    payload: PromoRAW[`id`]
  };
};


export default {
  loadFilms,
  loadPromo
};
