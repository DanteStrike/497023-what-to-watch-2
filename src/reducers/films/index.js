export {default as filmsTypes} from "./types.js";
export {default as filmsActions} from "./actions.js";
export {default as filmsSelectors} from "./selectors.js";
export {default as filmsReducer} from "./reducers.js";
