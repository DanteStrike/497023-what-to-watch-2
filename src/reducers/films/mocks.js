const filmsRAW = [
  {
    "id": 1,
    "name": `filmOne`,
    "poster_image": `img/filmOne-poster.jpg`,
    "preview_image": `img/filmOne.jpg`,
    "background_image": `img/filmOne.jpg`,
    "background_color": `#ffffff`,
    "video_link": `https://some-linkOne`,
    "preview_video_link": `https://some-linkOne`,
    "description": `Some text filmOne`,
    "rating": 1.0,
    "scores_count": 1,
    "director": `directorOne`,
    "starring": [`actorOne`, `actorTwo`],
    "run_time": 1,
    "genre": `genreOne`,
    "released": 2019,
    "is_favorite": false,
  },
  {
    "id": 3,
    "name": `filmTwo`,
    "poster_image": `img/filmTwo-poster.jpg`,
    "preview_image": `img/filmTwo.jpg`,
    "background_image": `img/filmTwo.jpg`,
    "background_color": `#000000`,
    "video_link": `https://some-linkTwo`,
    "preview_video_link": `https://some-linkTwo`,
    "description": `Some text filmTwo`,
    "rating": 2.0,
    "scores_count": 2,
    "director": `directorTwo`,
    "starring": [`actorThree`, `actorFour`],
    "run_time": 2,
    "genre": `genreTwo`,
    "released": 2020,
    "is_favorite": true,
  }
];

const adaptedFilmsData = [
  {
    id: 1,
    name: `filmOne`,
    posterImage: `img/filmOne-poster.jpg`,
    preview: {
      image: `img/filmOne.jpg`,
      videoSrc: `https://some-linkOne`
    },
    background: {
      image: `img/filmOne.jpg`,
      color: `#ffffff`
    },
    videoSrc: `https://some-linkOne`,
    description: `Some text filmOne`,
    rating: 1.0,
    scoresCount: 1,
    director: `directorOne`,
    starring: [`actorOne`, `actorTwo`],
    runTime: 1,
    genre: `genreOne`,
    released: 2019,
    isFavorite: false,
  }, {
    id: 3,
    name: `filmTwo`,
    posterImage: `img/filmTwo-poster.jpg`,
    preview: {
      image: `img/filmTwo.jpg`,
      videoSrc: `https://some-linkTwo`
    },
    background: {
      image: `img/filmTwo.jpg`,
      color: `#000000`
    },
    videoSrc: `https://some-linkTwo`,
    description: `Some text filmTwo`,
    rating: 2.0,
    scoresCount: 2,
    director: `directorTwo`,
    starring: [`actorThree`, `actorFour`],
    runTime: 2,
    genre: `genreTwo`,
    released: 2020,
    isFavorite: true,
  }
];

const normalizedFilmsData = {
  byIDs: {
    "1": adaptedFilmsData[0],
    "3": adaptedFilmsData[1]
  },
  allIDs: [1, 3]
};

const store = {
  films: {
    data: normalizedFilmsData
  }
};

export default {
  filmsRAW,
  adaptedFilmsData,
  normalizedFilmsData,
  store
};
