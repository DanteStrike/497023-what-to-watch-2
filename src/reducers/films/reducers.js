import types from "./types.js";
import {combineReducers} from "redux";

const filmsReducer = (state = {}, action) => {
  if (action.type === types.LOAD_FILMS) {
    return action.payload;
  }

  return state;
};

const reducer = combineReducers({
  data: filmsReducer
});

export default reducer;
