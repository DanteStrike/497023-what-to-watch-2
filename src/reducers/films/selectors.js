import {createSelector} from "reselect";

const getAllIDs = (store) => store.films.data.allIDs;
const getFilmsByID = (store) => store.films.data.byID;

const getCardsInfo = createSelector(
    [getAllIDs, getFilmsByID],
    (allIDs, films) => allIDs.map((filmID) => ({
      id: films[filmID].id,
      name: films[filmID].name,
      preview: films[filmID].preview,
    }))
);


export default {
  getAllIDs,
  getFilmsByID,
  getCardsInfo
};
