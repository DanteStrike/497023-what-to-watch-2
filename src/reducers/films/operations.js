import {filmsActions} from "./films";

const loadFilms = () => (dispatch, _, api) => {
  return api.get(`/films`)
    .then((response) => {
      dispatch(filmsActions.loadFilms(response.data));
    });
};

const loadPromo = () => (dispatch, _, api) => {
  return api.get(`/films/promo`)
    .then((response) => {
      dispatch(filmsActions.loadPromo(response.data));
    });
};

export default {
  loadFilms,
  loadPromo
};
