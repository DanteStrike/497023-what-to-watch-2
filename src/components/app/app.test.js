import React from "react";
import renderer from "react-test-renderer";
import {films} from "../../mocks/films.js";
import App from "./app.jsx";
import {Provider} from "react-redux";
import {createStore} from "redux";
import {reducer} from "../../reducer/reducer.js";

const store = createStore(reducer);

it(`render correctly App component`, () => {
  const component = renderer
    .create(
        <Provider store={store}>
          <App
            films={films}
          />
        </Provider>,
        {
          createNodeMock: (element) => {
            if (element.type === `video`) {
              return {
                src: null,
                isMuted: null,
                poster: null
              };
            }
            return null;
          }
        }
    ).toJSON();

  expect(component).toMatchSnapshot();
});
