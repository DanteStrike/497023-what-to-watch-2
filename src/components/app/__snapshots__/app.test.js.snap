// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly App component 1`] = `
<div
  className="player"
  onWheel={[Function]}
>
  <p
    style={
      Object {
        "color": "#eee5b5",
        "fontSize": "12px",
        "left": "5px",
        "margin": "0",
        "padding": "0",
        "position": "absolute",
        "top": "5px",
        "zIndex": "10",
      }
    }
  >
    <small
      style={
        Object {
          "fontSize": "12px",
        }
      }
    >
      Volume wheelUp(+) | wheelDown(-)
    </small>
    <strong
      style={
        Object {
          "display": "block",
          "fontSize": "16px",
        }
      }
    >
      20
       %
    </strong>
  </p>
  <video
    style={
      Object {
        "backgroundColor": "black",
        "display": "block",
        "height": "100%",
        "left": "50%",
        "minHeight": "100%",
        "minWidth": "100%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "100%",
      }
    }
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        DanteStrike
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
